### PR TITLE
Improving the Add Power Bill Screen

### PR DESCRIPTION
**Pull request by gabrielbmoro**
Some improvements such as:
- Snackbar message if the operation was successed or not;
- Loader progress;
- Reducing the code in the AddPowerBillScreen.

### CODE REVIEW

File: src/pocket_note/lib/ui/screens/resources/strings.dart [hunk 1]
```diff
@@ -31,4 +31,9 @@ final List<String> months = [
   "Outubro",
   "Novembro",
   "Dezembro"
-];
\ No newline at end of file
+];
+const String addPowerBillSuccessfulMessage = "Conta de luz salva com sucesso!";
+const String addPowerBillErrorMessageWithRetry = "Ocorreu um erro. Repita a operação!";
+const String addPowerBillErrorMessage = "Ocorreu um erro. Tente mais tarde!";
```
Comment (lucasrmoro) [inline src/pocket_note/lib/ui/screens/resources/strings.dart:36]: Do you not already have a generic error string?

File: src/pocket_note/lib/ui/screens/addpowerbill/store/add_power_bill_store.dart [hunk 2]
```diff
@@ -65,28 +65,15 @@ abstract class AddPowerBillBase with Store {
...
-  _onSaveSuccess(bool value) {
-    if (value) {
-      uiState = AddPowerBillUIState(
-        resultType: ResultType.success,
-      );
-    } else {
-      _onError();
+    try {
+      bool result = await savePowerBillUseCase.invoke(powerBill);
+      uiState = AddPowerBillUIState(isLoading: false);
```
Comment (lucasrmoro) [inline src/pocket_note/lib/ui/screens/addpowerbill/store/add_power_bill_store.dart]: The `uiState` copulation couldn't stay at finally block of try catch?

Comment (gabrielbmoro) [reply src/pocket_note/lib/ui/screens/addpowerbill/store/add_power_bill_store.dart]: It is an incredible suggestion. I am doing right now!! 💯

File: src/pocket_note/lib/ui/screens/resources/strings.dart (hunk 1 above)
Comment (gabrielbmoro) [reply src/pocket_note/lib/ui/screens/resources/strings.dart:36]: Currently we don't have, the idea is we have specific errors for each case (for now). I am not sure what will be the next steps

File: src/pocket_note/lib/data/local_data_source.dart [hunk 3]
```diff
@@ -19,6 +21,10 @@ class LocalDataSource {
   late Database _database;
 
   Future<void> init() async {
+    if (kIsWeb) {
```
Comment (gabrielbmoro) [inline src/pocket_note/lib/data/local_data_source.dart:24]: We need this to make the data base works on web browsers

File: src/pocket_note/lib/ui/screens/resources/strings.dart (hunk 1 above)
Comment (lucasrmoro) [reply src/pocket_note/lib/ui/screens/resources/strings.dart:36]: Ooh, I got it!! So LGTM hehe

**Review by lucasrmoro — APPROVED** [review]

**Pull request merged**